### DISCUSSION TITLE
TouchPad, MultiTouch, Apple Touchpad HID

### DISCUSSION THREAD

**Discussion by axife** [NicoHood/HID#292] [General]
I see library supports mouse including wheel.  But wheel is not smooth. It is more like scroll up/down buttons.

I wonder if there is HID that supports smooth  scrolling like apple's touch? (May be some windows touches has that)

Also I wonder if zooming support is possible with that HID library and 3 finger swipes.

**Comment by NicoHood** [comment]
I dont know what makes a scroll smooth. Maybe just scroll smaller values?
Zooming is not supported.

**Comment by axife** [comment]
So far I learnt in Windows 8.1 Microsoft introduced HID-based Precision Touchpad which aims to be a standard touchpad protocol.
https://www.reddit.com/r/windows/comments/1hky29/precision_touchpads_the_future_of_touchpads_on/

Quite helpful article with lots of links https://pavelfatin.com/scrolling-with-pleasure/#smooth-scrolling

Most touches our days are synaptics. By default (for BIOS at startup) they act as PS/2 mouse but with special sequence they are activated in full featured mode.  
https://www.aquaphoenix.com/hardware/ledlamp/reference/synaptics_touchpad_interfacing_guide.pdf

**Comment by teak** [comment]
+1. Found this library while looking to send gestures to a Windows or Linux device from any programmable USB device. Windows Precision Touchpad support sounds like what I'm looking for.
